### DISCUSSION TITLE
Node.js Binding

### DISCUSSION THREAD

**Discussion by twlite** [visioncortex/vtracer#71] [Show and tell]
I have made a package to access VTracer from nodejs (native binding) https://npm.im/@neplex/vectorizer

Installation is pretty straightforward

```sh
$ npm i @neplex/vectorizer
```

API is as minimal as possible

```js
import {
  vectorize,
  ColorMode,
  Hierarchial,
  PathSimplifyMode,
} from '@neplex/vectorizer';
import { readFile, writeFile } from 'node:fs/promises';

const src = await readFile('./raster.png');

const svg = await vectorize(src, {
  colorMode: ColorMode.Color,
  colorPrecision: 8,
  filterSpeckle: 4,
  spliceThreshold: 45,
  cornerThreshold: 60,
  hierarchical: Hierarchical.Stacked,
  mode: PathSimplifyMode.Spline,
  layerDifference: 6,
  lengthThreshold: 4,
  maxIterations: 2
});

console.log(svg); // <svg>...</svg>
await writeFile('./vector.svg', svg);
```

![image](https://github.com/visioncortex/vtracer/assets/46562212/8be6e7a3-71ac-42b1-b2f5-ac06fade5874)

Thanks for making this awesome project 🤗

**Comment by tyt2y3** [comment]
Thank you for sharing. If you have the time to contribute, I'd want to make an official binding, just like the one for Python.

**Reply by twlite** [reply]
@tyt2y3 sorry for the late response, I totally missed this (I guess I'm getting old now 😂). Sure, I am happy to transfer the repo here but I guess it'd be better to rename the package name :)

**Reply by tyt2y3** [reply]
ah yes, I would probably name it as `vtracer`, unless it's already taken?
well the valuable would be in the CI setup and the binding code!

**Reply by twlite** [reply]
`vtracer` seems to be [taken](https://npm.im/vtracer). You can create an org named `visioncortex` and name the package `@visioncortex/vtracer`
